datavault-webapp/src/main/java/org/datavaultplatform/webapp/controllers/admin/AdminRolesController.java

```diff
@@ -0,0 +1,307 @@
+package org.datavaultplatform.webapp.controllers.admin;
+
+import org.datavaultplatform.common.model.PermissionModel;
+import org.datavaultplatform.common.model.RoleModel;
+import org.datavaultplatform.common.model.RoleType;
+import org.datavaultplatform.webapp.services.RestService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import org.datavaultplatform.webapp.model.RoleViewModel;
+
+import java.util.*;
+//import java.util.stream.Collectors;
+
+
+@Controller
+public class AdminRolesController {
+
+    private static final Logger logger = LoggerFactory.getLogger(AdminRolesController.class);
+    private RestService restService;
+
+    public void setRestService(RestService restService) {
+        this.restService = restService;
+    }
+
+    private void setCommonModel(ModelMap model) {
+
+
+        List<RoleModel> roles = restService.getEditableRoles();
+
+        if (roles.isEmpty()) {
+            logger.info("getRolesListing: roles has no items in rest fetch");
+        }
+
+        model.addAttribute("roles", roles);
+
+
+        if (!model.containsAttribute("errormessage")) {
+            model.addAttribute("errormessage", "");
+        }
+
+
+    }
+
+
+    @RequestMapping(value = "/admin/roles", method = RequestMethod.GET)
+    public String getRolesListing(ModelMap model) {
+
+        try {
+                setCommonModel(model);
+
+        } catch (Exception e) {
```

Comment:
If an exception occurs part way through the common model values being set then we could end up with a weirdly rendered page (or possibly another exception coming out of freemarker) but this error handling looks more like debug code here so I think it may be possible to remove this completely. Also, we try not to catch `Exception` if possible because it will catch all exceptions whether they're what you think they are or not - a better approach if this is still needed would be to catch the specific type of exception that this is defending against.

Reply:
Catching all exceptions with current page setup will get generic error page, this is attempt to feed error back to the calling form without crashing the whole page,  trying to be more graceful to allow user to change issues in form and not lose form state.

Reply:
I've changed this to validate errors that we can predict and crash out when we can't. The forms are so small that I don't think we need to worry about losing form data when something unexpected occurs